File: 594-longest-harmonious-subsequence/longest-harmonious-subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        map<int,int> mp ;
        for(auto x :nums){
            mp[x]++;
        }
        int ans =0;
        for(auto x : mp){
            int y=x.first;
            if(mp.count(y+1)!=0){
               ans = max(ans,x.second+mp[y+1] );
            }

        }
        return ans;
    }
};

File: 594-longest-harmonious-subsequence/longest-harmonious-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findLHS = function (nums) {
    const mp = new Map();
    let ans = 0;
    for (let x of nums) {
        mp.set(x, (mp.get(x) || 0) + 1);
    }
    for (let [key, value] of mp.entries()) {
       let x = key;
       if(mp.has(x+1)){
        ans =Math.max(ans,value+mp.get(x+1));
       }
    }
    return ans;
};

